wallet-new/src/Cardano/Wallet/Kernel/NodeStateAdaptor.hs

```diff
@@ -488,28 +489,59 @@ waitForUpdate = liftIO . takeMVar =<< asks l
     l :: Res -> MVar ConfirmedProposalState
     l = ucDownloadedUpdate . view lensOf'
 
+
 -- | Get the difference between NTP time and local system time, nothing if the
 -- NTP server couldn't be reached in the last 30min.
 --
 -- Note that one can force a new query to the NTP server in which case, it may
 -- take up to 30s to resolve.
-defaultGetNtpDrift :: MonadIO m => TVar NtpStatus -> V1.ForceNtpCheck -> m V1.TimeInfo
-defaultGetNtpDrift tvar ntpCheckBehavior = liftIO $ do
-    when (ntpCheckBehavior == V1.ForceNtpCheck) $
-        atomically $ writeTVar tvar NtpSyncPending
-    mkTimeInfo <$> waitForNtpStatus
+defaultGetNtpDrift
+    :: MonadIO m
+    => TVar NtpStatus
+    -> V1.ForceNtpCheck
+    -> m V1.TimeInfo
+defaultGetNtpDrift tvar ntpCheckBehavior = liftIO $ mkTimeInfo <$>
+    if (ntpCheckBehavior == V1.ForceNtpCheck) then
+        forceNtpCheck >> getNtpOffset blockingLookupNtpOffset
```

Comment:
@KtorZ wouldn't be better to have `atomically` at the boundary but keep `forceNtpCheck` and `getNtpOffset` running in STM, so that they compose better?

Reply:
Well, we can't :upside_down_face:  ... We need to force the check first, commit the transaction, and then, get the offset :) 

On the other side, there's also a process watching this TVar which is triggered by the `forceNtpCheck`. Then, we just wait for that process to update the var to something that isn't "Pending"